gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystem.java

```diff
@@ -231,8 +231,9 @@ public WritableByteChannel create(URI path, CreateFileOptions createOptions) thr
               "Cannot create a file whose name looks like a directory: '%s'", resourceId));
     }
 
-    // Before creating a leaf directory we need to check if there are no conflicting files
-    // with the same name as any subdirectory
+    // Before creating a file we need to check:
+    // - if there are no conflicting files with the same name as any parent subdirectory
+    // - if there are no conflicting directory with the name as a file
```

Comment:
These comments are a bit confusing, what is "conflicting directory with the name as a file", is it file or directory? Should it be - check if there are no conflicting directories with the same name as the file being created?

Reply:
Clarified and added example.